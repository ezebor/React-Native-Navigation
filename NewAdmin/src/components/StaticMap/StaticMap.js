import React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, View} from 'react-native';
import MapView, {Marker, ProviderPropType} from 'react-native-maps';

import {PROVIDER_DEFAULT} from 'react-native-maps';

const StaticMap = ({region, provider, marker, dimensionsMap}) => {
  return (
    <View style={styles.container}>
      <MapView
        provider={provider}
        style={dimensionsMap}
        scrollEnabled={false}
        zoomEnabled={false}
        pitchEnabled={false}
        rotateEnabled={false}
        initialRegion={region}>
        {!!marker && (
          <Marker
            title={marker.title}
            description={marker.description}
            coordinate={marker.region}
          />
        )}
      </MapView>
    </View>
  );
};

StaticMap.propTypes = {
  provider: ProviderPropType,
  region: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    latitudeDelta: PropTypes.number,
    longitudeDelta: PropTypes.number,
  }).isRequired,
  marker: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    region: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      latitudeDelta: PropTypes.number,
      longitudeDelta: PropTypes.number,
    }),
  }),
  dimensionsMap: PropTypes.shape({
    width: PropTypes.number,
    height: PropTypes.number,
  }),
};

StaticMap.defaultProps = {
  marker: null,
  dimensionsMap: {
    width: 250,
    height: 250,
  },
  provider: PROVIDER_DEFAULT,
};

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  scrollview: {
    alignItems: 'center',
    paddingVertical: 40,
  },
});

export default StaticMap;
