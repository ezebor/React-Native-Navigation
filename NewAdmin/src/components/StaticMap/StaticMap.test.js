import React from 'react';
import {Dimensions} from 'react-native';
import renderer from 'react-test-renderer';

import StaticMap from './StaticMap';
const {width, height} = Dimensions.get('window');

const ASPECT_RATIO = width / height;
const LATITUDE = 37.78825;
const LONGITUDE = -122.4324;
const LATITUDE_DELTA = 0.0922;
const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;

const region = {
  latitude: LATITUDE,
  longitude: LONGITUDE,
  latitudeDelta: LATITUDE_DELTA,
  longitudeDelta: LONGITUDE_DELTA,
};

const marker = {
  title: 'Tiendanube',
  description: 'The best platform store',
  region,
};

const dimensionsMap = {width, height};

describe('renders correctly Static Map', () => {
  it('Default', () => {
    const tree = renderer
      .create(<StaticMap region={region} marker={marker} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('With custom dimensions', () => {
    const tree = renderer
      .create(
        <StaticMap
          region={region}
          marker={marker}
          dimensionsMap={dimensionsMap}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Without marker', () => {
    const tree = renderer.create(<StaticMap region={region} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
