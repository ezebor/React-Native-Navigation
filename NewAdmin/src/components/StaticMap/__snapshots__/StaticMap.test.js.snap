// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Static Map Default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View>
    <View />
  </View>
</View>
`;

exports[`renders correctly Static Map With custom dimensions 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View>
    <View />
  </View>
</View>
`;

exports[`renders correctly Static Map Without marker 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View />
</View>
`;
